src/library/scala/collection/ArrayOps.scala

```diff
@@ -249,8 +250,8 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
         case x: Array[Short]      => java.util.Arrays.copyOfRange(x, lo, hi)
         case x: Array[Boolean]    => java.util.Arrays.copyOfRange(x, lo, hi)
         case x: Array[Unit]       =>
-          val res = new Array[Unit](len)
-          Array.copy(xs, lo, res, 0, len)
+          val res = new Array[Unit](hi - lo)
```

Comment:
this never really happens - Array[Unit] is really a BoxedUnit array - see https://github.com/scala/scala/pull/6829/files

Reply:
This is inlining `len` which is `hi - lo` as long as `hi > lo` (the if check)

Reply:
it never actually gets to this case match

Reply:
I'm not familiar enough with that code to remove this match case. I'm just removing some bounds checks.